pkg/storage/cache.go

```diff
@@ -110,8 +110,16 @@ func GetInvalidIteratorCacheKey(storeID string) string {
 	return fmt.Sprintf("%s%s", invalidIteratorCachePrefix, storeID)
 }
 
-func GetInvalidIteratorByObjectRelationCacheKey(storeID, object, relation string) string {
-	return fmt.Sprintf("%s%s/%s#%s", invalidIteratorCachePrefix, storeID, object, relation)
+func GetInvalidIteratorByObjectRelationCacheKeys(storeID, object, relation string) []string {
+	return []string{fmt.Sprintf("%s%s-or/%s#%s", invalidIteratorCachePrefix, storeID, object, relation)}
+}
+
+func GetInvalidIteratorByUserObjectTypeCacheKeys(storeID string, users []string, objectType string) []string {
+	res := make([]string, 0, len(users))
+	for _, user := range users {
+		res = append(res, fmt.Sprintf("%s%s-otr/%s|%s", invalidIteratorCachePrefix, storeID, user, objectType))
+	}
```

Comment:
```suggestion
	res := make([]string, len(users))
	for i, user := range users {
		res[i] = fmt.Sprintf("%s%s-otr/%s|%s", invalidIteratorCachePrefix, storeID, user, objectType)
	}
```

We know the actual length of users, let's use it.